components/higher-order/with-dragging/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+
+const withDragging = ( OriginalComponent ) => {
+	class Draggable extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.onDragStart = this.onDragStart.bind( this );
+			this.onDragOver = this.onDragOver.bind( this );
+			this.onDragEnd = this.onDragEnd.bind( this );
+			this.cursorTop = null;
+			this.cursorLeft = null;
+			this.config = {
+				dragImageClass: 'components-with-dragging__invisible-drag-image',
+				cloneWrapperClass: 'components-with-dragging__clone',
+				cloneHeightTransformationBreakpoint: 700,
+				clonePadding: 20,
+			};
+			this.state = {
```

Comment:
Does this need to be state? Worth noting that `this.setState` _always_ incurs a re-render of the component. In this case, how we're using these IDs I don't think we need the render to occur.

Reply:
Nope, most definitely not - and it may be a cause of other problems too. Will definitely refactor this to avoid state updates.